Samples/BackgroundBlur/BackgroundBlur/BackgroundBlur.cpp

```diff
@@ -0,0 +1,515 @@
+// PARTICULAR PURPOSE.
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+#include "player.h"
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+#include <atlbase.h> // Contains the declaration of CComPtr.
+
+
+PCWSTR szTitle = L"BasicPlayback";
+PCWSTR szWindowClass = L"MFBASICPLAYBACK";
+
+HINSTANCE   g_hInstance;                    // current instance
+BOOL        g_bRepaintClient = TRUE;        // Repaint the application client area?
+CComPtr<CPlayer> g_pPlayer;                  // Global player object. 
+
+// Note: After WM_CREATE is processed, g_pPlayer remains valid until the
+// window is destroyed.
+
+// Forward declarations of functions included in this code module:
+BOOL                InitInstance(HINSTANCE, int);
+LRESULT CALLBACK    WndProc(HWND, UINT, WPARAM, LPARAM);
+INT_PTR CALLBACK    OpenUrlDialogProc(HWND, UINT, WPARAM, LPARAM);
+void                NotifyError(HWND hwnd, const WCHAR* sErrorMessage, HRESULT hr);
+void                UpdateUI(HWND hwnd, PlayerState state);
+HRESULT             AllocGetWindowText(HWND hwnd, WCHAR** pszText, DWORD* pcchLen);
+
+// Message handlers
+LRESULT             OnCreateWindow(HWND hwnd);
+void                OnFileOpen(HWND hwnd);
+void                OnOpenURL(HWND hwnd);
+void                OnStartStream(HWND hwnd);
+void                OnPlayerEvent(HWND hwnd, WPARAM pUnkPtr);
+void                OnPaint(HWND hwnd);
+void                OnResize(WORD width, WORD height);
+void                OnKeyPress(WPARAM key);
+
+// OpenUrlDialogInfo: Contains data passed to the "Open URL" dialog proc.
+struct OpenUrlDialogInfo
+{
+    WCHAR* pszURL;
+    DWORD cch;
+};
+
+int APIENTRY  wWinMain(HINSTANCE hInstance, HINSTANCE, PWSTR, int nCmdShow)
+{
+    HeapSetInformation(NULL, HeapEnableTerminationOnCorruption, NULL, 0);
+
+    MSG msg;
+
+    ZeroMemory(&msg, sizeof(msg));
+
+    // Perform application initialization.
+    if (!InitInstance(hInstance, nCmdShow))
+    {
+        NotifyError(NULL, L"Could not initialize the application.",
+            HRESULT_FROM_WIN32(GetLastError()));
+        return FALSE;
+    }
+
+    // Main message loop.
+    while (GetMessage(&msg, NULL, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+    }
+
+    // Clean up.
+    if (g_pPlayer)
+    {
+        g_pPlayer->Shutdown();
+    }
+    return 0;
+}
+
+//  Create the application window.
+BOOL InitInstance(HINSTANCE hInst, int nCmdShow)
+{
+    HWND hwnd;
+    WNDCLASSEX wcex;
+
+    g_hInstance = hInst; // Store the instance handle.
+
+    // Register the window class.
+    ZeroMemory(&wcex, sizeof(WNDCLASSEX));
+    wcex.cbSize = sizeof(WNDCLASSEX);
+    wcex.style = CS_HREDRAW | CS_VREDRAW;
+    wcex.lpfnWndProc = WndProc;
+    wcex.hInstance = hInst;
+    wcex.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1);
+    wcex.lpszMenuName = MAKEINTRESOURCE(IDC_MFPLAYBACK);
+    wcex.lpszClassName = szWindowClass;
```

Comment:
Nipick: In C++, unspecified members of a struct are initialized to zero, so:

```cpp
WNDCLASSEX wcex {};  // All members are initialized to zero, no need for ZeroMemory
```

Or even better:

```cpp
    WNDCLASSEX wcex {
        .cbSize = sizeof(WNDCLASSEX);
        .style = CS_HREDRAW | CS_VREDRAW;
        .lpfnWndProc = WndProc;
        .hInstance = hInst;
        .hbrBackground = (HBRUSH)(COLOR_WINDOW + 1);
        .lpszMenuName = MAKEINTRESOURCE(IDC_MFPLAYBACK);
        .lpszClassName = szWindowClass;
    };
    // These members are initialized, any remaining members are zero
```

Reply:
I can't use designated initializers (eg. the second option) with C++17, so I'll go with the first option.